client/dns_client.mli

```diff
@@ -78,8 +75,11 @@ sig
 
   type t
 
-  val create : ?size:int -> ?rng:(int -> Cstruct.t) -> ?nameserver:T.ns_addr -> clock:(unit -> int64) -> T.stack -> t
-  (** [create ~size ~rng ~nameserver ~clock stack] creates the state of the DNS client. *)
+  val create : ?size:int -> ?nameserver:T.ns_addr -> ?timeout:int64 -> T.stack -> t
+  (** [create ~size ~nameserver ~timeout stack] creates the state of the DNS client.
+      We use [timeout] (ns, default 3s) as a time budget for connect and request timeouts.
+      To specify a timeout, use [create ~timeout:(Duration.of_sec 5)].
+      *)
```

Comment:
The default IMHO is too small; on my machine I usually have more latency than that, would it harm to have 10s instead?

Reply:
Great point, we looked into this and we're gonna use 5s like in `resolve.h`.